config.go

```diff
@@ -67,25 +67,28 @@ func (c *Config) Load(target interface{}, modifiers ...TagModifier) error {
 		for i := 0; i < len(sourceFields); i++ {
 			targetFields[i].Field.Set(sourceFields[i].Field)
 		}
-	}
-
-	if err := loadError(errors); err != nil {
+	} else {
 		c.dumpSource()
-		return err
+		return newLoadError(errors)
 	}
 
 	chunk, err := dumpChunk(target)
 	if err != nil {
-		return fmt.Errorf("failed to serialize config (%s)", err.Error())
+		return fmt.Errorf("failed to serialize config: %w", err)
+	}
+
+	if err := c.postLoad(target); err != nil {
+		return newPostLoadError(err)
 	}
 
 	c.logger.Printf("Config loaded: %s", normalizeChunk(chunk))
+
 	return nil
 }
 
 // Call the PostLoad method of the given target if it conforms to
 // the PostLoadConfig interface.
-func (c *Config) PostLoad(target interface{}) error {
```

Comment:
Assuming that the callers in the main nacelle repo will need to be removed as we do this automatically?

Might be worth a note in the changelog as well as this changes the public interface.

Reply:
That's a good point. Considering semver, that technically makes this v3.0.0 of config. Are you opposed to that? I'd rather keep to proper semver than keeping the "main" version of the nacelle repo (which I would consider the "nacelle version" as a whole) than breaking it just to keep the major versions the same on all the repos.

Reply:
Yeah go ahead and bump it. That probably means we'll bump the main nacelle repo to 3 as well since the major version bumps are transitive?

I'm cool with that if it makes sense.

Reply:
Do they have to be transitive? Couldn't the nacelle repo stay at v2 and reference config v3?

Reply:
Only if the API surface change is relevant to end users of the nacelle repo, which I guess isn't true in this case.

Major version bump here, minor in the main repo?

Reply:
Yeah, that's what I was thinking.